### PR TITLE
Event-emitter cheat sheet

### PR DESCRIPTION
**Pull request by alexalikiotis**

### CODE REVIEW

File: event-emitter.adoc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Same as the `myEmitter.on()` method.
+
+```js
+myEmitter.addListener('welcome', (data) => console.log(data.msg));
+```
+
+==== EventEmitter.prototype.once()
+
+Listen to a specific event only once.
```
Comment (lholmquist) [inline event-emitter.adoc:52]: Maybe expand on why this is important?

File: event-emitter.adoc [hunk 2]
```diff
@@ -0,0 +1,109 @@
+= Node.js Event Emitter Cheat Sheet
+
+This cheat sheet covers the basic principles around node.js event emitters.
+
+== Event Emitter
+
+Node.js allows us to create and handle custom events easily by using events module. Event module includes EventEmitter class which can be used to raise and handle custom events.
```
[the remark this answers is not in the document]
Comment (alexalikiotis) [reply event-emitter.adoc:7]: Thanks! That's way better than what I wrote.

File: event-emitter.adoc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+Removes all listeners, or those of the specified event.
+
+```js
+myEmitter.removeAllListeners(['connection']); // receives an array of events
+```
+
+==== EventEmitter.prototype.setMaxListeners()
+
+Sets the maximum allowed listeners for a specific event. Node.js will print
+a warning if more than 10 listeners are added for a specific event.
```
[the remark this answers is not in the document]
Comment (BethGriggs) [reply event-emitter.adoc:83]: I think 'modify limit' is slightly more appropriate than suppress? Perhaps:


> Sets the maximum allowed listeners for a specific event. By default, Node.js will print a warning if more than 10 listeners are added for a specific event. `setMaxListeners` allows you to modify this limit.

File: event-emitter.adoc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```js
+myEmitter.removeAllListeners(['eventOne']); // receives an array of events
+```
+
+==== EventEmitter.prototype.setMaxListeners()
+
+Sets the maximum allowed listeners for a specific event. Node.js will print
+a warning if more than 10 listeners are added for a specific event.
+`setMaxListeners` allows you to suppress this warning.
+
```
Comment (BethGriggs) [inline event-emitter.adoc:85]: Is it also worth mentioning `events.defaultMaxListeners` and the order of precedence between `setMaxListeners` and `events.defaultMaxListeners`? https://nodejs.org/api/events.html#events_events_defaultmaxlisteners

**Review by lholmquist — APPROVED** [review]

**Pull request merged**